athena-neptune/src/main/java/com/amazonaws/athena/connectors/neptune/NeptuneMetadataHandler.java

```diff
@@ -0,0 +1,249 @@
+
+    public NeptuneMetadataHandler() {
+        super(false, SOURCE_TYPE);
+        neptuneConnection = new NeptuneConnection(System.getenv("neptune_endpoint"), System.getenv("neptune_port"));
+        // g = neptuneConnection.getTraversalSource();
+        glue = getAwsGlue();
+    }
+
+    @VisibleForTesting
+    protected NeptuneMetadataHandler(AWSGlue glue,
+            // GetTablesRequest glueReq,
+            NeptuneConnection neptuneConnection, EncryptionKeyFactory keyFactory, AWSSecretsManager awsSecretsManager,
+            AmazonAthena athena, String spillBucket, String spillPrefix) {
+        super(glue, keyFactory, awsSecretsManager, athena, SOURCE_TYPE, spillBucket, spillPrefix);
+        this.glue = glue;
+        // this.glueReq = glueReq;
+        this.neptuneConnection = neptuneConnection;
+    }
+
+    /**
+     * Since the entire Neptune cluster is considered as a single graph database,
+     * just return the glue database name provided as a single database (schema)
+     * name.
+     * 
+     * @param allocator Tool for creating and managing Apache Arrow Blocks.
+     * @param request   Provides details on who made the request and which Athena
+     *                  catalog they are querying.
+     * @return A ListSchemasResponse which primarily contains a Set<String> of
+     *         schema names and a catalog name corresponding the Athena catalog that
+     *         was queried.
+     * @see GlueMetadataHandler
+     */
+    @Override
+    public ListSchemasResponse doListSchemaNames(BlockAllocator allocator, ListSchemasRequest request) {
+        logger.info("doListSchemaNames: enter - " + request);
+
+        Set<String> schemas = new HashSet<>();
+        schemas.add(glue_database_name);
+        return new ListSchemasResponse(request.getCatalogName(), schemas);
+    }
+
+    /**
+     * Used to get the list of tables that this data source contains. In this case,
+     * fetch list of tables in the Glue database provided.
+     * 
+     * @param allocator Tool for creating and managing Apache Arrow Blocks.
+     * @param request   Provides details on who made the request and which Athena
+     *                  catalog and database they are querying.
+     * @return A ListTablesResponse which primarily contains a List<TableName>
+     *         enumerating the tables in this catalog, database tuple. It also
+     *         contains the catalog name corresponding the Athena catalog that was
+     *         queried.
+     * @see GlueMetadataHandler
+     */
+    @Override
+    public ListTablesResponse doListTables(BlockAllocator allocator, ListTablesRequest request) {
+        logger.info("doListTables: enter - " + request);
+
+        List<TableName> tables = new ArrayList<>();
+        GetTablesRequest getTablesRequest = new GetTablesRequest();
+
+        if (glue_database_name != null) {
+            getTablesRequest.setDatabaseName(glue_database_name);
+        }
```

Comment:
I think we should use the database name from the ListTablesRequest (aka schemaName). Otherwise it may confuse customers because they will list tables in database_y but get tables from database_x... then when they query database_y they will be confused as to why they get errors about no such table.

Reply:
Makes sense, doListTables is always called in context of a schema populated by calling doListSchemaNames.

Reply:
Done